src/analyzer.rs

```diff
@@ -9,437 +9,503 @@ use std::collections::{BTreeMap, HashMap};
+        let mut members = vec![];
+        //debug!("analyzing object {}", serde_json::to_string(&schema).unwrap());
+        let reqs = schema.required.clone().unwrap_or_default();
+        for (key, value) in props {
+            let value_type = value.type_.clone().unwrap_or_default();
+            let rust_type = match value_type.as_ref() {
+                "object" => {
+                    let mut dict_key = None;
+                    if let Some(additional) = &value.additional_properties {
+                        dict_key = self.resolve_additional_properties(additional, stack, key, value)?;
+                    } else if value.properties.is_none()
+                        && value.x_kubernetes_preserve_unknown_fields.unwrap_or(false)
+                    {
+                        dict_key = Some("serde_json::Value".into());
+                    }
+                    if let Some(dict) = dict_key {
+                        format!("BTreeMap<String, {}>", dict)
+                    } else {
+                        format!("{}{}", stack, key.to_upper_camel_case())
+                    }
                 }
-                if let Some(dict) = dict_key {
-                    format!("BTreeMap<String, {}>", dict)
-                } else {
-                    format!("{}{}", stack, key.to_upper_camel_case())
+                "string" => {
+                    if let Some(_en) = &value.enum_ {
+                        trace!("got enum string: {}", serde_json::to_string(&schema).unwrap());
+                        format!("{}{}", stack, key.to_upper_camel_case())
+                    } else {
+                        "String".to_string()
+                    }
                 }
-            }
-            "string" => {
-                if let Some(_en) = &value.enum_ {
-                    trace!("got enum string: {}", serde_json::to_string(&schema).unwrap());
-                    format!("{}{}", stack, key.to_upper_camel_case())
-                } else {
-                    "String".to_string()
+                "boolean" => "bool".to_string(),
+                "date" => extract_date_type(value)?,
+                "number" => extract_number_type(value)?,
+                "integer" => extract_integer_type(value)?,
+                "array" => {
+                    // recurse through repeated arrays until we find a concrete type (keep track of how deep we went)
+                    let (mut array_type, recurse_level) =
+                        self.array_recurse_for_type(value, stack, key, 1)?;
+                    trace!("got array {} for {} in level {}", array_type, key, recurse_level);
+                    if use_k8s_openapi_condition && key == "conditions" && is_conditions(&value) {
+                        array_type = "Vec<k8s_openapi::apimachinery::pkg::apis::meta::v1::Condition>".into();
+                    } else {
+                        array_recurse_level.insert(key.clone(), recurse_level);
+                    }
```

Comment:
Nice. Tiny nit; we should probably use `Vec<Condition>` and add `use k8s_openapi::apimachinery::pkg::apis::meta::v1::Condition;` at the top level if it's found (you can have a `Container::uses_conditions()` fn to gate this import)

Reply:
done in https://github.com/kube-rs/kopium/pull/204/commits/47f67f21aa3d39f888cce7444a976e76211b4b50

Reply:
i used the qualified name instead because didn't want any potential conflicts with other imports, so this seemed safer. but since this already behind a flag, i guess its safe to use a top level import.